docs/documentation/guides/adding-a-new-payment-processor.md

```diff
@@ -0,0 +1,449 @@
+
+You can implement a Payment Method class if you want to add specific payment method logic:
+```python
+# payment_processors.py
+# ...
+from .models.payment_methods import CustomPaymentMethod
+
+class TriggeredPaymentProcessor(PaymentProcessorBase, TriggeredProcessorMixin):
+    payment_method_class = CustomPaymentMethod
+    #...
+```
+
+```python
+# models/payment_methods.py
+# ...
+from silver.models import PaymentMethod
+
+class CustomPaymentMethod(PaymentMethod):
+    class Meta:
+        proxy = True
+    
+    @property
+    def token(self):
+        return self.decrypt_data(self.data.get('token'))
+
+    @token.setter
+    def token(self, value):
+        self.data['token'] = self.encrypt_data(value)  # self.encrypt_data is already implemented
+```
+
+You might have noticed that earlier, when we implemented `execute_transaction`, we mentioned a token. 
+Most payment processors will provide you with a token, because you are not supposed to handle the
+customer's payment method info, so you will use the token to refer to a payment method.
+
+The token is useful if the customer wants to save his payment method info for further or recurring 
+payments. Read about how you can obtain it in the [payment operation section](#the-payment-operation).
+
+
+### The payment operation
+
+This section tries to explain how the customer's payment operation usually goes like.
+
+1. The customer usually clicks a Pay button somewhere in a different application, which will trigger a
+[create request on the Silver's `/transactions` endpoint](Transactions#create-a-transaction).
+2. The transaction is created, and the response payload contains a field called `pay_url`, which
+is a Silver URL to a view that will act as a payment page, which most likely you'll also have to 
+implement because of the differences between the payment processors. You can read more about the 
+payment page [here](#the-payment-page)
+3. The App redirects the customer to the `pay_url`. There the customer can enter it's payment method
+info and accept the transaction. Keep in mind the customer might choose to cancel the operation, enter
+wrong or expired payment method data etc.. In some cases the customer might be redirected from the 
+Silver payment page to the external Payment Processor payment page.
+4. Depending on the outcome of the operation, the customer may be redirected to the 
+`complete_payment_view` Silver view, which will call the `handle_transaction_response` method of our
+`TriggeredPaymentProcessor` class. Sometimes the PaymentProcessor won't rely on the customer to access
+the complete_payment_view (since it is not a guarantee it will happen) and will send a webhook containing
+info about the external transaction.
+5. From there the client can be redirected to a final `redirect_url` or have a Silver template rendered
+stating the state of the transaction.
+
+Step 4. is important because the `handle_transaction_response` method is where we can update our
```

Comment:
Maybe is more readable to have it right after step 4

Reply:
I'll redo this part :+1:.